cmd/skywire-cli/commands/log/root.go

```diff
@@ -210,13 +242,93 @@ func download(ctx context.Context, log *logging.Logger, httpC http.Client, targe
 	file, _ := os.Create(pubkey + "/" + fileName) //nolint
 	defer file.Close()                            //nolint
 
-	if err := dmsgget.Download(ctx, log, &httpC, file, target, maxSize); err != nil {
+	if err := Download(ctx, log, &httpC, file, target, maxSize); err != nil {
 		log.WithError(err).Errorf("The %s for visor %s not available", fileName, pubkey)
 		return err
 	}
 	return nil
 }
 
+// Download downloads a file from the given URL into 'w'.
+func Download(ctx context.Context, log logrus.FieldLogger, httpC *http.Client, w io.Writer, urlStr string, maxSize int64) error {
+	req, err := http.NewRequest(http.MethodGet, urlStr, nil)
+	if err != nil {
+		log.WithError(err).Fatal("Failed to formulate HTTP request.")
+	}
+	resp, err := httpC.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to connect to HTTP server: %w", err)
+	}
+	if resp.StatusCode == http.StatusOK {
+		// 200 OK
+		if maxSize > 0 {
+			if resp.ContentLength > maxSize*1024 {
+				return fmt.Errorf("requested file size is more than allowed size: %d KB > %d KB", (resp.ContentLength / 1024), maxSize)
+			}
+		}
+		n, err := CancellableCopy(ctx, w, resp.Body, resp.ContentLength)
+		if err != nil {
+			return fmt.Errorf("download failed at %d/%dB: %w", n, resp.ContentLength, err)
+		}
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				log.WithError(err).Warn("HTTP Response body closed with non-nil error.")
+			}
+		}()
+		return nil
+	}
+	// Convert the non-200 status code to an error
+	return &httpError{Status: resp.StatusCode}
+}
+
+type readerFunc func(p []byte) (n int, err error)
+
+func (rf readerFunc) Read(p []byte) (n int, err error) { return rf(p) }
+
+// CancellableCopy will call the Reader and Writer interface multiple time, in order
```

Comment:
Do we just need the progress reader here? I am pretty sure the io.Copy is already copying in chunks.

Reply:
This part of the implementation goes back to dmsgget and I think it should be optional or at least be optional to silence the download progress writer.

I know why it was done that way previously. Because if it doesn't print anything and there were no errors, you can't easily tell what it's doing or if it's stuck.

Considering that `skywire-cli log` can run for 15 minutes __with the default flags__ or even longer if you try to collect multiple previous day's transport bandwidth logs, it's probably a good idea to print progress.

A flag to disable that would be nice, but I think it should be implemented first for `dmsgget` and then we can update `skywire-cli log` to implement the same method.